File: Java/Program/RAM.java
public class RAM extends Komponen {
    private int kapasitas;
    private int kecepatan;
    private String jenis;

    public RAM() {
    }

    public RAM(int kapasitas, int kecepatan, String jenis, String nama, float harga, String merk) {
        super(nama, harga, merk);
        this.kapasitas = kapasitas;
        this.kecepatan = kecepatan;
        this.jenis = jenis;
    }

    public void setKapasitas(int kapasitas) {
        this.kapasitas = kapasitas;
    }

    public void setKecepatan(int kecepatan) {
        this.kecepatan = kecepatan;
    }

    public void setJenis(String jenis) {
        this.jenis = jenis;
    }

    public int getKapasitas() {
        return kapasitas;
    }

    public int getKecepatan() {
        return kecepatan;
    }

    public String getJenis() {
        return jenis;
    }

    public void upgrade() {
        kapasitas *= 2;
        System.out.println("RAM upgraded to " + kapasitas + " GB");
    }

    @Override
    public void getInfo() {
        super.getInfo();
        System.out.println("Kapasitas: " + kapasitas + " GB, Kecepatan: " + kecepatan + " MHz, Jenis: " + jenis);
    }
}

File: Python/Program/RAM.py
from Komponen import Komponen

class RAM(Komponen):
    def __init__(self, kapasitas=0, kecepatan=0, jenis="", nama="", harga=0, merk=""):
        super().__init__(nama, harga, merk)
        self.kapasitas = kapasitas
        self.kecepatan = kecepatan
        self.jenis = jenis
    
    def set_kapasitas(self, kapasitas):
        self.kapasitas = kapasitas
    
    def set_kecepatan(self, kecepatan):
        self.kecepatan = kecepatan
    
    def set_jenis(self, jenis):
        self.jenis = jenis
    
    def get_kapasitas(self):
        return self.kapasitas
    
    def get_kecepatan(self):
        return self.kecepatan
    
    def get_jenis(self):
        return self.jenis
    
    def upgrade(self):
        self.kapasitas *= 2
        print(f"RAM upgraded to {self.kapasitas} GB")
    
    def get_info(self):
        super().get_info()
        print(f"Kapasitas: {self.kapasitas} GB, Kecepatan: {self.kecepatan} MHz, Jenis: {self.jenis}")

